version_updates.md

```diff
@@ -0,0 +1,120 @@
+same value as the 'AvailableVersion' on the OpenStackVersion resources. This
+triggers the OpenStackVersion to reconcile and will update the images
+on the associated OpenStackControlplane and OpenStackDataplane resources in
+that namespace.
+
+```shell
+$ oc get openstackversion
+NAME              TARGET VERSION   AVAILABLE VERSION
+openstack         1.0.0            1.0.1
+```
+
+```yaml
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackVersion
+metadata:
+  labels:
+  name: openstack
+spec:
+  openstackControlPlaneName: openstack
+  targetVersion: 1.0.1
+```
+
+## Custom images for Cinder Volume
+
+In order to allow administrators to configure customized Cinder volume backends
+with custom containerImages for each backend you must exclude those images
+from automatic updates. This is because there is no way to fully automate
+the creation of those images at this time. To exclude Cinder volume named
+'netapp' from automatic updates you can add the name 'netapp' to the
+'cinderVolumeExcludes' array on the OpenStackVersion CR. Multiple named
+Cinder volume backends may be excluded.
+
+When the custom built containerImage(s) for a Cinder Volume have been
+created you can deploy them by editing/patching/updating the
+OpenStackControlplane directly.
+
+
+```yaml
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackVersion
+metadata:
+  labels:
+  name: openstack
+spec:
+  openstackControlPlaneName: openstack
+  targetVersion: 1.0.1
+  cinderVolumeExcludes:
+    -netapp
+```
+
+## Custom Images for Neutron
+
+TBD
+
+## Custom images for other OpenStack services
+
+In general administrators should not need to customize containerImages
+for any other OpenStack services images. The exception would be
+the need to apply a hotfix. In order to accomidate a hotfixed or
+patched containerImage you can update the 'serviceExcludes'
+array on the OpenStackVersion to contain the camel cased
+name for the services that needs to be excluded from
+automatic image updates. For example if you need
+to hotfix the Glance API service container image you could
+add 'GlanceApi' to 'serviceExcludes' and then proceed
+with editing/patching/updating the OpenStackControlplane
+with a customized version of this service.
+
+## Custom ordering of service updates
+
+Container image updates currently occur in parallel for a given CR.
+First the Controlplane will be updated. Then any Dataplane CRs.
+If a situation arises where we need to update one service before
+another we can use OpenStackVersion to drive this.
+
+## Things currently not supported
+
+ * Rolling back to a prior openstack version directly via OpenStackVersion.
+   A rollback would require the administrator to install an old CSV,
+   and patch various CRs to update the containerImages among other things.
+ * Major upgrades are not targeted at this time. However the OpenStackVersion
+   CR may be useful as a means to help drive those as well.
```

Comment:
Does any CRD changes mean a major update?

Reply:
Right now any change would be a minor update. In the future for major upgrades (should we choose to reuse this resource for that workflow) we could have a SemVer check so that major changes go to one reconcile function and minor another

Reply:
Sorry I think I asked the wrong question. Do we support CRD changes during the minor upgrade solution proposed here?

Reply:
I don't see why not so long as those changes don't modify the containerImage's themselves.

Reply:
ack